text/0000-intl-v2.md

```diff
@@ -0,0 +1,218 @@
+        * No additional service is required for this to function.
+    * Localization
+        * Gramatticus, our well stablished legacy platform for localization, will remain as the service that powers the translations.
+
+Cons:
+
+    * Complexity:
+        * LWC Compiler must be revamped.
+        * New language to be integrated with code editors, and linters to be introduced.
+        * New metadata to be added to LWC Compilation to describe the label and entities dependencies.
+    * External Dependency:
+        * Using fluent, as the primary language to describe messages is tricky because we don't control the destiny of that project.
+
+## Adoption strategy
+
+### Backward Compatibility & Risks
+
+One of the compelling things about this proposal is that it does not affect in any way the existing HTML Parser, or syntax. It does not introduce any new directive either. In previous attempts to solve this problem, that was one of the major concerns. This new feature should be backward compatibility, and users of the framework can continue using the existing label resolution in platform alongside the new feature.
+
+### Prior Art
+
+FormatJS, which relies of helpers and template literals to define the messages, and format the data. Our own implementation of the label system today, which promotes the import of a platform label, and the manual transformation of such string value to produce the final result. As a result, that component is just not localizable anymore.
+
+## How we teach this
+
+For LWC Authors, this should be easy to learn. They will have to learn few things:
+
+1. That there is a new `this.i18n` value available for anyone extending `LightningElement`. The values there are computed messages, and they change at the same time the UI gets updated, and of course, it can be used from a template, the same way you use any other field, or value from the component.
+2. A new syntax, the fluent syntax, must be learned. There is plenty of documentation on the web for fluent.
+3. The same data that is available to the template, is also available to the fluent file.
+4. How to use platform labels inside their fluent files, and this is mostly a convention system since it relies on the import syntax provided by fluent.
+
+Once they understand there 4 things, they should be able to create very compelling, fully localizable components.
+
+## Open questions
+
+1. Q: How to use messages in a for-loops using the context of the loop as data?
+
+### 2. Q: Do we need a reverse parser? from Fluent to Grammaticus? if someone were going to write a label as a Fluent message, how do we make sure this can be put through the grammaticus engine for translations?
+
+Caridy: I don't think that should be the case for 1P, certain, we can lint 1P to make sure that everything that they use in the fluent file is either interpolation of existing entities or labels, or direct usage of entities and labels. For 2P/3P, the question remains.
```

Comment:
What is the downside of dropping Grammaticus support entirely? In other words, what do we gain by adding Grammaticus support?

Based on my read of [this post](https://engineering.salesforce.com/open-sourcing-grammaticus-308d4c9f696e/), Grammaticus is not directly exposed to component authors today; it is just used internally.

Reply:
That is the thing... for Admins, the UI to localize their app is backed up by the Grammaticus system in Java land. I will love to drop it entirely, but that's not going to happen. Keep in mind that neither Aura, LWR nor LWC has been able to integrate it yet.